### PR TITLE
✨ Look up official EKS AMI when appropriate

### LINKED ISSUE #1778: Support official EKS AMI lookup
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

Once EKS support is implemented for worker nodes, we'd like to be able to lookup and consume the official EKS AMIs. 

Per docs [here](https://docs.aws.amazon.com/eks/latest/userguide/retrieve-ami-id.html) there is a way to retrieve this using SSM:
```
aws ssm get-parameter --name /aws/service/eks/optimized-ami/1.16/amazon-linux-2/recommended/image_id --region region-code --query "Parameter.Value" --output text
```

I think it would be reasonable to use this api from within CAPA to look up these AMIs.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

Open to suggestions for how to toggle between the community AMIs and the EKS AMIs, but some options are:
1. Automatic if using the EKS control plane provider
2. Boolean spec field on AWSMachine / AWSMachineTemplate
3. Annotation on AWSMachine / AWSMachineTemplate
4. some combination like 1/2 or 1/3

### PR DESCRIPTION
**Pull request by rudoi**
**What this PR does / why we need it**:
This adds a lookup for the official EKS AMI.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1778 

This is WIP until #1724 goes in. Will be using those types to determine if using the EKS AMI is appropriate. I'll also need some additional changes when #1810 goes in. 

Also still working on finding the right IAM updates. I'm having an obnoxiously difficult time getting a policy statement that is properly scoped. Wildcards are not appropriate at all for SSM Parameter Store.

### CODE REVIEW

File: pkg/cloud/services/ec2/instances.go [hunk 1]
```diff
@@ -738,6 +745,12 @@ func (s *Service) DetachSecurityGroupsFromNetworkInterface(groups []string, inte
 	return nil
 }
 
+func (s *Service) shouldUseEksAMI() bool {
+	gvk := s.scope.InfraCluster().GetObjectKind().GroupVersionKind()
+
+	return gvk.Kind == "AWSManagedCluster"
```
Comment (detiber) [inline pkg/cloud/services/ec2/instances.go:751]: Should this just be `EKSCluster` instead of `AWSManagedCluster`?

Comment (rudoi) [reply pkg/cloud/services/ec2/instances.go:751]: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1724 introduces `AWSManagedCluster` and `AWSManagedControlPlane`. Though after deciding to put most of the logic in the controlplane provider, this should be `AWSManagedControlPlane` actually.

**Comment by rudoi** [comment]
@detiber the SSM code path is only used when there's an AWSManagedControlPlane in play, so I don't expect any IAM errors when using a non-EKS cluster. 

This PR needs #1724 anyway, so I think we hold this until it's in and I can test both code paths at once.

/hold

**Comment by ncdc** [comment]
/retitle [WIP] ✨ Look up official EKS AMI when appropriate

File: cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go [hunk 2]
```diff
@@ -147,6 +147,15 @@ func (t Template) controllersPolicy() *iamv1.PolicyDocument {
 					"secretsmanager:TagResource",
 				},
 			},
+			{
```
Comment (randomvariable) [inline cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go:150]: After you've rebased, this should go behind the EKS flags.

**Comment by randomvariable** [comment]
Looks fine other than the rebasing.

File: cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go (hunk 2 above)
Comment (rudoi) [reply cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go:150]: Any precedence for feature flags in clusterawsadm yet?

Comment (randomvariable) [reply cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go:150]: Sorry, misspoke. I meant the flag in terms of `t.Spec.ClusterAPIControllers.EKS.Enable` at https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go#L149 than a `-- features` thing

Comment (rudoi) [reply cmd/clusterawsadm/cloudformation/bootstrap/cluster_api_controller.go]: done

**Comment by randomvariable** [comment]
```
=== RUN   Test_RenderCloudformation
--- FAIL: Test_RenderCloudformation (0.04s)
    template_test.go:118: Differing output (default):
        AWSTemplateFormatVersion: 2010-09-09
        Resources:
          AWSIAMInstanceProfileControlPlane:
            Properties:
skipped 257 lines unfold_more
                    - ec2.amazonaws.com
                Version: 2012-10-17
              RoleName: nodes.cluster-api-provider-aws.sigs.k8s.io
            Type: AWS::IAM::Role
        
FAIL
FAIL	sigs.k8s.io/cluster-api-provider-aws/cmd/clusterawsadm/cloudformation/bootstrap	0.111s
```

**Comment by rudoi** [comment]
Ok, fixed up those template tests.

**Pull request merged**